sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -211,9 +214,31 @@ public HttpHeaders getHeaders() {
 
         @Override
         public Mono<String> getBodyAsString() {
-            return bodyIntern().aggregate().asString().doFinally(s -> {
-                if (!reactorNettyConnection.isDisposed()) {
-                    reactorNettyConnection.channel().eventLoop().execute(reactorNettyConnection::dispose);
+            return getBodyAsByteArray().map(bytes -> {
```

Comment:
Both JDK client and Netty client has to do the same logic for converting response byte array to string. It would be better to put this in Core utils somewhere to reduce duplication and if there are any fixes or updates to this logic, we don't need to update in two places.

Reply:
Yeah I was thinking about that, also would give the opportunity to test is more directly and thoroughly.